### PR TITLE
 Initial introduction of generated api client

### PR DESCRIPTION
**Pull request by jbowes**
Supports only identity and catalog now, but its a start!

### CODE REVIEW

**Comment by tim-speed** [comment]
What's the benefit of this?
Just moving the specs out of Marketplace to this shared library?
Looks fine to me.

**Comment by jbowes** [comment]
> What's the benefit of this?

`zz_oag_generated_identity.go` and `zz_oag_generated_catalog.go`, which provide an api for speaking to those two services. And maybe i'm biased, but it's a pretty nice API, too!

File: Gopkg.toml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+
```
Comment (jelmersnoeck) [inline Gopkg.toml]: 👍

Comment (jelmersnoeck) [reply Gopkg.toml]: Maybe remove the example comments?

File: specs/catalog.yaml [hunk 2]
```diff
@@ -0,0 +1,894 @@
...
+produces:
+- application/json
+consumes:
+- application/json
+securityDefinitions:
+  jwtRequired:
+    type: apiKey
+    name: Authorization
+    in: header
+  anonymous:
```
Comment (jelmersnoeck) [inline specs/catalog.yaml]: Do we still need this here, even for external clients?

File: specs/connector.yaml [hunk 3]
```diff
@@ -0,0 +1,556 @@
+swagger: "2.0"
```
Comment (jelmersnoeck) [inline specs/connector.yaml]: Is there a reason we have this in here but not the oag file?

**Comment by tim-speed** [comment]
Totally biased, but I agree it's a pretty nice API :)
Like Jelmer mentioned, are we going to get a `zz_oag_generated_marketplace.go`?

**Comment by jbowes** [comment]
> Like Jelmer mentioned, are we going to get a zz_oag_generated_marketplace.go?

yes, but not yet. needs https://github.com/jbowes/oag/issues/10 first :(

File: Gopkg.toml (hunk 1 above)
Comment (jbowes) [reply Gopkg.toml]: will do!

**Comment by jbowes** [comment]
> It looks like not all specs get generated files. What's the reason for putting it in the repo if it's not generated?

Just so its easier to take the specs out of our internal repo and bring them over here. That's also why some things are in the specs that shouldn't be/don't need to be: its a bug in the original spec's `audience`, or a bug in our processor.

**Comment by jbowes** [comment]
marketplace added, too!

**Pull request merged**